pkg/egressgateway/policy.go

```diff
@@ -157,37 +161,65 @@ func (gwc *gatewayConfig) deriveFromPolicyGatewayConfig(gc *policyGatewayConfig)
 
 	switch {
 	case gc.iface != "":
-		// If the gateway config specifies an interface, use the first IPv4 assigned to that
-		// interface as egress IP
+		// If the gateway config specifies an interface, use the first IPv4/v6 assigned to that
+		// interface as egress IPs
 		gwc.ifaceName = gc.iface
-		gwc.egressIP, err = netdevice.GetIfaceFirstIPv4Address(gc.iface)
+		gwc.egressIP4, err = netdevice.GetIfaceFirstIPv4Address(gc.iface)
 		if err != nil {
-			gwc.egressIP = EgressIPNotFoundIPv4
+			gwc.egressIP4 = EgressIPNotFoundIPv4
 			return fmt.Errorf("failed to retrieve IPv4 address for egress interface: %w", err)
 		}
+
+		if gc.v6needed {
+			gwc.egressIP6, err = netdevice.GetIfaceFirstIPv6Address(gc.iface)
+			if err != nil {
+				gwc.egressIP6 = EgressIPNotFoundIPv6
+				return fmt.Errorf("failed to retrieve IPv4 address for egress interface: %w", err)
+			}
+		}
 	case gc.egressIP.IsValid():
 		// If the gateway config specifies an egress IP, use the interface with that IP as egress
 		// interface
-		gwc.egressIP = gc.egressIP
+		gwc.egressIP4 = gc.egressIP
```

Comment:
Nit: might want to leave a comment here to state the specifying an IPv6 is not supported yet